packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -87,6 +92,48 @@ const capturePhaseEvents = new Set([
   TOP_WAITING,
 ]);
 
+const isArray = Array.isArray;
+
+function dispatchEventsForPlugins(
+  topLevelType: DOMTopLevelEventType,
+  eventSystemFlags: EventSystemFlags,
+  nativeEvent: AnyNativeEvent,
+  targetInst: null | Fiber,
+  rootContainer: Element | Document,
+): void {
+  const nativeEventTarget = getEventTarget(nativeEvent);
+  const syntheticEvents = [];
+
+  for (let i = 0; i < plugins.length; i++) {
+    const possiblePlugin: PluginModule<AnyNativeEvent> = plugins[i];
```

Comment:
why isn't plugins typed as Array<PluginModule<AnyNativeEvent>>? could avoid the truthy check. or am I missing how it's used?

Reply:
PluginModule is nullish, so we still have to check for that.

Reply:
It's because the array has an empty first plugin that isn't included in normal builds (responder plugin), so 0 is undefined.